server.go

```diff
@@ -74,12 +85,11 @@ type Server struct {
 }
 
 // NewFromConfig creates a new veneur server from a configuration specification.
-func NewFromConfig(conf Config) (ret Server, err error) {
+func NewFromConfig(conf Config, transport http.RoundTripper) (ret Server, err error) {
```

Comment:
Hey @aditya-stripe,

Having to pass a `RoundTripper` to `NewFromConfig` feels wonky, but it's the best way I can find to do it since it uses the `HttpClient` as it starts the server for the initial read from Consul… ideas for how to fix that or is it ok?